yt_dlp/extractor/pstream.py

```diff
@@ -0,0 +1,70 @@
+from .common import InfoExtractor
+from ..utils import base64
+
+
+class PStreamIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?pstream\.net/(e|v)/(?P<id>[a-zA-Z0-9]+)'
+    _TESTS = [{
+        'url': 'https://www.pstream.net/v/YdX2V1RLJjJVayY',
+        'md5': '5487d5137cfeb078efb13454cd538f77',
+        'info_dict': {
+            'id': 'YdX2V1RLJjJVayY',
+            'ext': 'mp4',
+            'title': 'How I Keep My House In Order',
+            'thumbnail': 'https://i.pstream.net/l8sR7XK3DHszbtL4_vbWzA/YdX2V1RLJjJVayY/hrKo81Kbxb0iPXt21bL4Coshcp8S4DzO.jpg',
+        }
+    },
+        {
+        'url': 'https://www.pstream.net/v/mQN5YQKD6jLXvRa',
+        'md5': '08245a58b410a0c62896dd758d16e127',
+        'info_dict': {
+            'id': 'mQN5YQKD6jLXvRa',
+            'ext': 'mp4',
+            'title': 'DanMachi_S01E10_MULTi_1080p_10bits_BluRay_x265_AAC_-Punisher694.mkv',
+            'thumbnail': None,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        url = f'https://www.pstream.net/e/{video_id}'
+
+        webpage = self._download_webpage(url, video_id)
```

Comment:
Inline `url` in `webpage`

Reply:
I think it's important to not inline it, when debugging, or just exploring the script, see that url is always changed by the embed version...